core/controllers/topic_editor.py

```diff
@@ -149,8 +151,43 @@ def get(self, topic_id: str) -> None:
         additional_story_summary_dicts = [
             summary.to_dict() for summary in additional_story_summaries]
 
+        canonical_stories_ids = [summary['id'] for
+            summary in canonical_story_summary_dicts]
+        canonical_stories = list(filter(
+            None, story_fetchers.get_stories_by_ids(canonical_stories_ids)))
+        canonical_stories_dict: Dict[str, story_domain.Story] = {}
+        for canonical_story in canonical_stories:
+            canonical_stories_dict.update({
+                canonical_story.id: canonical_story})
         updated_canonical_story_summary_dicts = []
+
         for summary in canonical_story_summary_dicts:
+            story = canonical_stories_dict[summary['id']]
```

Comment:
This will raise a KeyError if `summary['id'] is not in the dictionary. You were catching these cases before

Reply:
Done